### PR TITLE
python310Packages.find-libpython: init at 0.3.0

### PR DESCRIPTION
**Pull request by jleightcap**
###### Description of changes

> Finds the libpython associated with the current Python environment, wherever it may be hiding.

https://github.com/ktbarrett/find_libpython

Packaged to un-break the cocotb package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/find-libpython/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+}:
+
+buildPythonPackage rec {
+  pname = "find_libpython";
```
Comment (fabaff) [inline pkgs/development/python-modules/find-libpython/default.nix:7]:
```suggestion
  pname = "find-libpython";
```

Should be [normalized](https://nixos.org/manual/nixpkgs/stable/#contributing).

File: pkgs/development/python-modules/find-libpython/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  version = "0.3.0";
+  format = "setuptools";
```
Comment (fabaff) [inline pkgs/development/python-modules/find-libpython/default.nix:9]: We should stick to supported Python releases. The change is high that the module will not be used/tests on old Python releases.

```suggestion
  format = "setuptools";

  disabled = pythonOlder "3.7";
```

File: pkgs/development/python-modules/find-libpython/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-bn/l2a9/rW3AZstVFaDpyQpx8f6yuy+OTNu0+DJ26eU=";
+  };
+
```
Comment (fabaff) [inline pkgs/development/python-modules/find-libpython/default.nix:15]: The upstream repo contains [tests](https://github.com/ktbarrett/find_libpython/tree/master/tests). 

Please add `pythonImportsCheck`.

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Pull request merged**